addons/web/static/src/xml/notification.xml

```diff
@@ -11,8 +11,8 @@
                 <span class="d-inline" aria-hidden="true">&amp;times;</span>
             </button>
         </t>
-        <div t-if="widget.title" class="toast-header">
-            <span t-attf-class="fa fa-2x mr-3 #{widget.icon} o_notification_icon"
+        <div t-if="widget.title" class="toast-header" t-att-style="widget.titleColor ? _.str.sprintf('color:%s', widget.titleColor) : ''">
+            <span t-attf-class="fa fa-2x mr-3 #{widget.headerIconDisable ? 'd-none' : widget.icon} o_notification_icon"
```

Comment:
Do not use `d-none`, instead add a `t-if` around the element to avoid rendering it.